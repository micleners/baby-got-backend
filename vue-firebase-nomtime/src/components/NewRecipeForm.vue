<script>
import { mapMutations } from "vuex";
import { db } from "../main";
export default {
  name: "NewRecipeForm",
  data() {
    return {
      recipe: {},
      recipeTitle: "",
      recipeTime: ""
    };
  },
  methods: {
    ...mapMutations(["ADD_RECIPE"]),
    showRecipes(query) {
      console.log(query);
      this.$store.dispatch("getRecipes", query);
    },
    addRecipe: function() {
      const recipe = {};
      console.log(this.recipeTitle);
      console.log(this.readyInMinutes);
      db.collection("recipes").add({
        recipe: { label: this.recipeTitle, totalTime: this.recipeTime }
      });
      this.recipe = {};
    }
  }
};
</script>

<template>
  <div>
    <form @submit.prevent="addRecipe">
      Name:
      <input
        class="recipe-title custom"
        type="text"
        placeholder="Recipe Name"
        v-model="recipeTitle"
      />
      Duration:
      <input
        class="recipe-time custom"
        type="number"
        placeholder="How long it takes to cook (minutes)"
        v-model.number="recipeTime"
      />
      <button type="submit" class="add-recipe-button">Submit</button>
    </form>
    <button @click="showRecipes(recipeTitle)">Select This Plan</button>
  </div>
</template>
