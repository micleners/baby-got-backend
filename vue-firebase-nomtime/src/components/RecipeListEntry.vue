<script>
import { db } from "../main";
import Confirm from "@/components/Confirm";
export default {
  name: "RecipeListEntry",
  components: {
    Confirm
  },
  props: {
    recipe: {
      type: Object,
      required: true,
      defualt: () => ({})
    }
  },
  methods: {
    deleteRecipe(id) {
      this.$refs.confirm
        .open("Delete this recipe?", "", { color: "red" })
        .then(confirm => {
          console.log(confirm);
          if (confirm) {
            db.collection("recipes")
              .doc(id)
              .delete();
          }
        });
    }
  }
};
</script>

<template>
  <div>
    <div class="recipe-entry-box">
      <button @click="deleteRecipe(recipe.id)" class="remove">Delete</button>
      <a :href="recipe.recipe.url">
        <div>
          <div class="recipe-entry-title">{{ recipe.recipe.label }}</div>
          <div>Ready In: {{ recipe.recipe.totalTime }} minutes</div>
        </div>
      </a>
    </div>
    <confirm ref="confirm"></confirm>
  </div>
</template>

<style scoped>
.remove {
  float: right;
  text-transform: uppercase;
  font-size: 0.8em;
  background: #f9d0e3;
  border: none;
  padding: 5px;
  color: #ff0076;
  cursor: pointer;
  margin: 8px;
}

.recipe-entry-title {
  font-weight: bold;
}

a {
  text-decoration: none;
  color: #2c3e50;
}

.recipe-entry-box {
  padding: 10px;
  background: white;
  transition: background-color 0.5s ease;
  border-bottom: 0.25px solid #f8f8f8;
}

.recipe-entry-box:hover {
  background: #d8e0d7;
}
</style>