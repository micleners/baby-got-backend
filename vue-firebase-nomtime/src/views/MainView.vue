<script>
import { mapMutations } from "vuex";
import RecipeListEntry from "@/components/RecipeListEntry";
import NewRecipeForm from "@/components/NewRecipeForm";
import LoadRecipeForm from "@/components/LoadRecipeForm";
import { db } from "../main";

export default {
  name: "MainView",
  components: {
    RecipeListEntry,
    NewRecipeForm,
    LoadRecipeForm
  },
  data() {
    return {
      recipes: [],
      recipe: {},
      recipeTitle: "",
      recipeTime: "",
      formMode: "add"
    };
  },
  firestore() {
    return {
      recipes: db.collection("recipes")
    };
  },
  methods: {
    ...mapMutations(["ADD_RECIPE"]),
    addRecipe: function() {
      const recipe = {};
      recipe["title"] = this.recipeTitle;
      recipe["readyInMinutes"] = this.recipeTime;
      console.log(recipe);
      this.ADD_RECIPE(recipe);
      this.recipe = {};
    }
  }
};
</script>

<template>
  <div class="main">
    <div class="actions"></div>
    <div class="left">
      <h2>Recipe List</h2>
      <ul>
        <li v-for="(recipe, index) in recipes" v-bind:key="index">
          <RecipeListEntry :recipe="recipe" />
        </li>
      </ul>
    </div>
    <div class="right">
      <div v-on:click="formMode = 'find'">
        <img
          v-bind:class="{open: formMode === 'find'}"
          class="dropdown-triangle"
          src=".././assets/triangle.svg"
        />
        <h3>Find Recipes</h3>
      </div>
      <div v-show="formMode === 'find'">
        <LoadRecipeForm />
      </div>
      <hr />
      <div v-on:click="formMode = 'add'">
        <img
          v-bind:class="{open: formMode === 'add'}"
          class="dropdown-triangle"
          src=".././assets/triangle.svg"
        />
        <h3>Create New Recipe</h3>
      </div>
      <div v-show="formMode === 'add'">
        <NewRecipeForm />
      </div>
    </div>
  </div>
</template>

<style>
html,
#app,
.home {
  height: 100%;
}

body {
  background-color: #f4f4f4;
  margin: 0;
  height: 100%;
}

.main {
  display: grid;
  grid-template-columns: repeat(2, 50%);
  grid-template-rows: auto 1fr;
  grid-template-areas:
    "left actions"
    "left right";
  height: 100%;
}

input.custom {
  border: none;
  padding: 15px;
  width: calc(100% - 40px);
  box-shadow: 0 5px 5px lightgrey;
  margin-bottom: 10px;
  outline: none;
}

.right {
  text-align: left;
  padding: 30px;
}

.left {
  text-align: left;
  grid-area: left;
  overflow: scroll;
}

ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

h2 {
  background: #2c3d50;
  color: #f9f9f9;
  padding: 10px 20px;
  margin: 0;
  box-sizing: border-box;
  height: 50px;
}

h3 {
  display: inline;
  vertical-align: middle;
}

.right {
  grid-area: right;
  background-color: #e9e9e9;
}

.actions {
  grid-area: actions;
  height: 50px;
  margin: 0px;
  padding: 0px;
  background: #8a929c;
}

.add-recipe-button {
  box-shadow: 0 5px 5px lightgrey;
  float: right;
  text-transform: uppercase;
  background: #2c3d50;
  border: none;
  padding: 5px;
  margin: 10px;
  color: #f9f9f9;
  cursor: pointer;
}

.dropdown-triangle {
  transform: rotate(0deg);
  transition: transform 0.5s linear;
  vertical-align: middle;
  margin: 10px;
  width: 10px;
  height: 10px;
}

.dropdown-triangle.open {
  transform: rotate(90deg);
  transition: transform 0.5s linear;
}
</style>
