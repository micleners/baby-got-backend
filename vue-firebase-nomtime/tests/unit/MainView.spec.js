import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import MainView from "@/views/MainView";
import RecipeListEntry from "@/components/RecipeListEntry";
import NewRecipeForm from "@/components/NewRecipeForm";
import initialState from "@/store/state";
import recipesFixture from "./fixtures/recipes";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("MainView", () => {
  let state;

  const build = () => {
    const wrapper = shallowMount(MainView, {
      localVue,
      store: new Vuex.Store({ state })
    });

    return {
      wrapper,
      recipeListEntry: () => wrapper.find(RecipeListEntry),
      newRecipe: () => wrapper.find(NewRecipeForm)
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });

  it("renders the component", () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    // arrange
    const { recipeListEntry, newRecipe } = build();

    // assert
    expect(recipeListEntry().exists()).toBe(true);
    expect(newRecipe().exists()).toBe(true);
  });

  it("passes a binded recipe prop to the entry component", () => {
    // arrange
    const { wrapper, recipeListEntry } = build();
    wrapper.setData({
      recipes: [
        {
          title: "Veggie Delight"
        }
      ]
    });

    // assert
    expect(recipeListEntry().vm.recipes).toBe(wrapper.vm.recipes);
  });

  it("passes binded recipes prop to entry component", () => {
    // arrange
    state = recipesFixture;
    const { recipeListEntry } = build();

    // assert
    expect(recipeListEntry().vm.recipes).toBe(state.recipes);
  });
});
