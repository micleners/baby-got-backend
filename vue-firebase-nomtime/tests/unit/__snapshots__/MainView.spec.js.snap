// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView renders the component 1`] = `
<div class="main">
  <div class="left">
    <recipelistentry-stub recipes="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"></recipelistentry-stub>
  </div>
  <div class="right">
    <newrecipeform-stub></newrecipeform-stub>
  </div>
</div>
`;
